frontend/src/metabase/admin/tasks/containers/JobInfoApp.jsx

```diff
@@ -64,40 +56,25 @@ const renderJobsTable = jobs => {
   );
 };
 
-class JobInfoApp extends Component {
-  async componentDidMount() {
-    try {
-      const info = (await this.props.fetchJobInfo()).payload;
-      this.setState({
-        scheduler: info.scheduler,
-        jobs: info.jobs,
-        error: null,
-      });
-    } catch (error) {
-      this.setState({ error });
-    }
-  }
-
-  render() {
-    const { children } = this.props;
-    const { error, scheduler, jobs } = this.state || {};
-
-    return (
-      <LoadingAndErrorWrapper loading={!scheduler} error={error}>
-        <JobInfoRoot>
-          <JobInfoHeader>
-            <AdminHeader title={t`Scheduler Info`} />
-          </JobInfoHeader>
-          {renderSchedulerInfo(scheduler)}
-          {renderJobsTable(jobs)}
-          {
-            // render 'children' so that the invididual task modals show up
-            children
-          }
-        </JobInfoRoot>
-      </LoadingAndErrorWrapper>
-    );
-  }
-}
+export const JobInfoApp = ({ children }) => {
+  const { data, error, isFetching } = useGetTasksInfoQuery();
 
-export default connect(null, { fetchJobInfo })(JobInfoApp);
+  return (
+    <LoadingAndErrorWrapper
+      loading={isFetching && !data?.scheduler}
```

Comment:
I don't think wee need `!data?.scheduler` check now that we have proper `isFetching` state.

```suggestion
      loading={isFetching}
```